pkg/scheduler/algorithm/predicates/metadata_test.go

```diff
@@ -1243,6 +1243,547 @@ func TestGetTPMapMatchingSpreadConstraints(t *testing.T) {
 	}
 }
 
+func TestPodSpreadMap_addPod(t *testing.T) {
+	tests := []struct {
+		name              string
+		preemptorPod      *v1.Pod
+		addedPod          *v1.Pod
+		existingPods      []*v1.Pod
+		nodeIdx           int // denotes which node 'addedPod' belongs to
+		nodes             []*v1.Node
+		injectPodPointers map[topologyPair][]int // non-negative index refers to existingPods[i], negative index refers to addedPod
```

Comment:
This is hard to understand. Maybe `wantPodInTopology`. Although, this might go away with the count based map, so we could leave it for now.